codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustReservedWords.kt

```diff
@@ -33,34 +44,19 @@ class RustReservedWordSymbolProvider(private val base: RustSymbolProvider) : Wra
         val reservedWordReplacedName = internal.toMemberName(shape)
         val container = model.expectShape(shape.container)
         return when {
-            container is StructureShape -> when (baseName) {
-                "build" -> "build_value"
-                "builder" -> "builder_value"
-                "default" -> "default_value"
-                "send" -> "send_value"
-                // To avoid conflicts with the `make_operation` and `presigned` functions on generated inputs
-                "make_operation" -> "make_operation_value"
-                "presigned" -> "presigned_value"
-                "customize" -> "customize_value"
-                // To avoid conflicts with the error metadata `meta` field
-                "meta" -> "meta_value"
-                else -> reservedWordReplacedName
+            container is StructureShape -> when (val mapped = reservedWordConfig.structMemberMap[baseName]) {
+                null -> reservedWordReplacedName
+                else -> mapped
             }
 
-            container is UnionShape -> when (baseName) {
-                // Unions contain an `Unknown` variant. This exists to support parsing data returned from the server
-                // that represent union variants that have been added since this SDK was generated.
-                UnionGenerator.UnknownVariantName -> "${UnionGenerator.UnknownVariantName}Value"
-                "${UnionGenerator.UnknownVariantName}Value" -> "${UnionGenerator.UnknownVariantName}Value_"
-                else -> reservedWordReplacedName
+            container is UnionShape -> when (val mapped = reservedWordConfig.unionMemberMap[baseName]) {
+                null -> reservedWordReplacedName
+                else -> mapped
             }
 
-            container is EnumShape || container.hasTrait<EnumTrait>() -> when (baseName) {
-                // Unknown is used as the name of the variant containing unexpected values
-                "Unknown" -> "UnknownValue"
-                // Real models won't end in `_` so it's safe to stop here
-                "UnknownValue" -> "UnknownValue_"
-                else -> reservedWordReplacedName
+            container is EnumShape || container.hasTrait<EnumTrait>() -> when (val mapped = reservedWordConfig.enumMemberMap[baseName]) {
```

Comment:
`EnumTrait` is only applicable to string shapes; a member shape's container cannot be a string shape.
```suggestion
            container is EnumShape -> when (val mapped = reservedWordConfig.enumMemberMap[baseName]) {
```

Reply:
Hmmm... Yeah, I think this is dead code. `toMemberName` isn't even called for enum rendering. I think I'll keep it around in the event that someone refactors EnumGenerator to use `toMemberName` though since this is technically correct.